src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -318,7 +318,7 @@ public static void RegisterType (string java_class, Type t)
 					if (jniFromType != java_class) {
 						managedToJni.Add (t, java_class);
 					}
-				} else {
+				} else if (!ignoreExistingRegistration) {
```

Comment:
Which made me think that we could also hide this warning under a log category possibly.